src/main/kotlin/blackjack/domain/card/DefaultCardFactory.kt

```diff
@@ -0,0 +1,26 @@
+package blackjack.domain.card
+
+import blackjack.domain.Card
+import blackjack.domain.CardShape
+import blackjack.domain.CardType
+
+class DefaultCardFactory : CardFactory {
+    override fun createCards(): MutableList<Card> {
+        return createAllShapeCards().shuffled().toMutableList()
+    }
+
+    private fun createAllShapeCards(): MutableList<Card> {
+        val cards = mutableListOf<Card>()
+        CardShape.values().forEach { cardShape ->
+            createSingleShapeCardsOf(cardShape, cards)
+        }
+        return cards
+    }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
    private fun createAllShapeCards(): List<Card> {
        return CardShape.values().flatMap { cardShape ->
            CardType.values().map { cardType -> Card(cardShape, cardType) }
        }
    }
```